File: tests/machine/x/clj/list_set_ops.clj
(ns main)

(defn _union_all [a b]
  (vec (concat a b)))
(defn _union [a b]
  (vec (distinct (concat a b))))
(defn _except [a b]
  (vec (remove (set b) a)))
(defn _intersect [a b]
  (vec (distinct (filter (set b) a))))
(defn -main []
  (println (_union [1 2] [2 3]))
  (println (_except [1 2 3] [2]))
  (println (_intersect [1 2 3] [2 4]))
  (println (count (_union_all [1 2] [2 3])))
)

(-main)

File: tests/machine/x/fs/list_set_ops.fs
open System

printfn "%s" (String.concat " " (List.map string [1; 2] union [2; 3]))
printfn "%s" (String.concat " " (List.map string [1; 2; 3] except [2]))
printfn "%s" (String.concat " " (List.map string [1; 2; 3] intersect [2; 4]))
printfn "%A" (List.length [1; 2] union [2; 3])
